data/src/main/java/com/linkedin/data/schema/generator/DefaultSampleDataCallback.java

```diff
@@ -135,7 +135,12 @@ public ByteString getFixed(String fieldName, FixedDataSchema schema)
   public String getEnum(String fieldName, EnumDataSchema enumDataSchema)
   {
     List<String> symbols = enumDataSchema.getSymbols();
-    return symbols.get(nonNegative(symbols.size() - 1));
+    // enum without any symbols is allowed, return "EmptyEnum" as a reference in doc.
+    if (symbols.size() < 1)
+    {
+      return "EmptyEnum";
+    }
+    return symbols.get(symbols.size() - 1);
```

Comment:
Do you want to keep using nonNegative to generate the index?
then I think you can just 
```
    if (symbols.size() < 1)
    {
      symbols = Arrays.asList("EmptyEnum");
    }

```

Reply:
I think we should keep the behavior same as previously, randomly get one symbol from the symbol list. When the symbol list size < 1, I think we should directly return the value to avoid unnecessary randomly generating index step. Updated the code to add nonNegative back.